### PR TITLE
RMT support for Ruby 2.7

### PR DESCRIPTION
**Pull request by lcaparroz**
## Why?

The `rmt-server` is not currently building for Factory/Tumbleweed on OBS, which can't decide which package should provide the `bundler` for Ruby 2.6.

After further investigation, we noticed OBS, in fact, sets the default Ruby version to 2.7 on Factory and Tumbleweed, which is not 100% compatible without some tweaking.

## What?

We bumped some gems versions (`config:1.7.2` is incompatible with Ruby 2.7), changed method calls which generated warnings, and disabled "deprecated" warnings, generated mostly by Rails 5.1.

### CODE REVIEW

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/110491/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 565a6222d1d8d8128b036a36270d2f59e9685afd into b7219a4c03ecaf2d9dd2d5ddc3bbe466f38edc57)

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/110495/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 565a6222d1d8d8128b036a36270d2f59e9685afd into b7219a4c03ecaf2d9dd2d5ddc3bbe466f38edc57)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/110508/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 565a6222d1d8d8128b036a36270d2f59e9685afd into b7219a4c03ecaf2d9dd2d5ddc3bbe466f38edc57)

File: package/obs/rmt-server.spec [hunk 1]
```diff
@@ -22,10 +22,14 @@
...
-%define ruby_version %{rb_default_ruby_suffix}
+
+# Only build for the distribution default Ruby version
+%define rb_build_versions     %{rb_default_ruby}
+%define rb_build_ruby_abis    %{rb_default_ruby_abi}
+%define ruby_version          %{rb_default_ruby_suffix}
 
 Name:           rmt-server
-Version:        2.5.8
+Version:        2.5.9
```
Comment (ikapelyukhin) [inline package/obs/rmt-server.spec:32]: #569 has already claimed this version number, and we've actually deployed 2.5.9 to our servers. :thinking:

Comment (lcaparroz) [reply package/obs/rmt-server.spec]: Bumped to 2.5.10 on 2fb5e5c.

**Comment by suse-tests-pass** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The version of RMT is not consistent. These files must specify the same version:</p>
<ul>
  <li><code>lib/rmt.rb</code> specifies <code>2.5.9</code></li>
  <li><code>package/obs/rmt-server.spec</code> specifies <code>2.5.10</code></li>
<</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by suse-tests-fail** [comment]
Prophet reports failure.
https://ci.suse.de/job/scc-RMT-integration-tests/110761/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment.
Failed fast in project rmt on step docker-compose-run-tests. (Merged 2fb5e5ce68845fe28b124157e759da31065ed5b4 into b7219a4c03ecaf2d9dd2d5ddc3bbe466f38edc57)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/110762/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 220b4e34d7af231cdd8777d1d34312bbb50fc3f4 into b7219a4c03ecaf2d9dd2d5ddc3bbe466f38edc57)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/110996/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 6871732e499c437468278c4db258ca54d4b572b9 into b7219a4c03ecaf2d9dd2d5ddc3bbe466f38edc57)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/111051/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 90b8126dfd1caded483356039a5a0562ee66fc4b into 06eb40f6ea695f0075d2afddbfb0c6872ef9c5ed)

**Pull request merged**